+neuron/Neuron.m

```diff
@@ -321,12 +321,12 @@ function reset_sections()
                             clib.neuron.section_unref(section);
                         else
                             % Valid section: append.
-                            all_sections{end+1} = neuron.Section(section, owner);
+                            all_sections{end+1} = neuron.Section(section, owner, all_sections{1}.mech_list, all_sections{1}.range_list, true);
```

Comment:
I have no intuitiion for wy the new section object would be copying information from the 1st section?

Reply:
Currently the reason allsec is slow is partly due to every section getting all neuron functions and splitting them into mech list and range list:
![image](https://github.com/mcdougallab/matlabneuroninterface/assets/144685504/d953b063-6c50-42df-90b4-e669394036e1)

This PR adds two params to the Section constructor to pass these values if they should not be calculated again.